src/coreclr/vm/syncblk.cpp

```diff
@@ -494,11 +471,30 @@ SyncBlockCache::SyncBlockCache()
         INJECT_FAULT(COMPlusThrowOM());
     }
     CONTRACTL_END;
-}
 
+    m_pCleanupBlockList = NULL;
+    m_FreeBlockList     = NULL;
+
+    // NOTE: CRST_UNSAFE_ANYMODE prevents a GC mode switch when entering this crst.
+    // If you remove this flag, we will switch to preemptive mode when entering
+    // g_criticalSection, which means all functions that enter it will become
+    // GC_TRIGGERS.  (This includes all uses of LockHolder around SyncBlockCache::GetSyncBlockCache().
+    // So be sure to update the contracts if you remove this flag.
+    ::new (&m_CacheLock) Crst(CrstSyncBlockCache, (CrstFlags) (CRST_UNSAFE_ANYMODE | CRST_DEBUGGER_THREAD));
```

Comment:
Could you please change the type of `m_CacheLock` to be `CrstStatic` and initialize it using `Init` method?

(As written, the compiler would add global initialization of `m_CacheLock` using static constructor/destructor that would lead to intermittent crashes during shutdown.)

Reply:
OK, thanks